includes/Story_Renderer/HTML.php

```diff
@@ -111,82 +72,45 @@ public function __construct( Story $story ) {
 	 * @return string The complete HTML markup for the story.
 	 */
 	public function render() {
-		$markup = self::DEFAULT_DOCTYPE . $this->story->get_markup();
-		$markup = $this->adapt_encoding( $markup );
+		$markup = $this->story->get_markup();
 		$markup = $this->replace_html_head( $markup );
-		$markup = $this->remove_noscript_amp_boilerplate( $markup );
 
-		$this->document = $this->load_html( $markup );
+		// TODO: What if transformation failed?
+		$document = Document::fromHtml( $markup, get_bloginfo( 'charset' ) );
+
+		// This  should never actually happen.
+		if ( ! $document ) {
+			wp_die(
+				esc_html__( 'There was an error generating the web story, probably because of a server misconfiguration. Try contacting your hosting provider or open a new support request.', 'web-stories' ),
+				esc_html__( 'Web Stories', 'web-stories' ),
+				[
+					'response'  => 500,
+					'link_url'  => esc_url( __( 'https://wordpress.org/support/plugin/web-stories/', 'web-stories' ) ),
+					'link_text' => esc_html__( 'Visit Support Forums', 'web-stories' ),
+				]
+			);
```

Comment:
To facilitate unit testing, perhaps this should pass the `'exit' => false` arg, and output buffer the `wp_die()` call to then use it as the return value?

```suggestion
			ob_start();
			wp_die(
				esc_html__( 'There was an error generating the web story, probably because of a server misconfiguration. Try contacting your hosting provider or open a new support request.', 'web-stories' ),
				esc_html__( 'Web Stories', 'web-stories' ),
				[
					'response'  => 500,
					'link_url'  => esc_url( __( 'https://wordpress.org/support/plugin/web-stories/', 'web-stories' ) ),
					'link_text' => esc_html__( 'Visit Support Forums', 'web-stories' ),
					'exit' => false,
				]
			);
			return ob_get_clean();
```